data-model/src/main/java/io/micronaut/data/model/query/builder/AbstractSqlLikeQueryBuilder.java

```diff
@@ -1085,7 +1130,7 @@ private void appendCriteriaForOperator(StringBuilder whereClause,
             boolean shouldEscape = propertyPath.shouldEscape();
 
             boolean[] needsTrimming = {false};
-            traversePersistentProperties(propertyPath.getAssociations(), propertyPath.getProperty(), (associations, property) -> {
+            traversePersistentProperties(propertyPath.getAssociations(), propertyPath.getProperty(), true, (associations, property) -> {
```

Comment:
Looks like overkill adding a new parameter. Just refactor this code and extract: `traverseForCriteriaPropertyPath(propertyPath, lambda)`

Reply:
This parameter is used for different traversal of properties. For Azure Cosmos, how I thought it could/should work, we don't traverse through embedded properties but select whole embedded field since this is how (de)serialization works. Except for where and composite id, then need to traverse them all for comparison `(where settlementid.code = @p1 and settlementid.country.id == @p2)`

Reply:
Yeah, so just extract traverse methods based on the use-case so they can be modified in the parent class.

Reply:
Alright, done in https://github.com/micronaut-projects/micronaut-data/pull/1790/commits/5fd5483386094fd1bb4fe0142823020b53a59dee